src/components/MoneyRequestConfirmationList.tsx

```diff
@@ -301,6 +302,15 @@ function MoneyRequestConfirmationList({
 
     const shouldShowTax = isTaxTrackingEnabled(isPolicyExpenseChat, policy, isDistanceRequest) && !isTypeInvoice;
 
+    const transactionCurrency = TransactionUtils.getCurrency(transaction);
+    useEffect(() => {
+        if (isEmptyObject(policy) || !shouldShowTax || !taxRates) {
+            return;
+        }
+        const defaultTaxCode = TransactionUtils.getDefaultTaxCode(policy, transactionCurrency);
+        IOU.setMoneyRequestTaxRate(transactionID, defaultTaxCode ?? '');
```

Comment:
Changing the function `IOU.setMoneyRequestTaxRate` to receive only the currency instead of the whole transaction object allow us to only watch the `currency` instead of the `transaction` object, which is what we want to react to if it changes.